tripal/tripal
# Tv4g2 issue2036 tripaldbx extension not installed

# Bug Fix

### Issue # 2036

## Description

This change removes all traces of `tripaldbx.settings` by moving the respective config lines into Tripal itself.

This change is necessary because it prevents sites from being able to import full site configurations that had been generated from sites that still had `tripaldbx.settings` records. 

## Testing?



1. Build a completely new Docker with this branch†
2. Export a full site configuration using the Configuration Manager: `/admin/config/development/configuration/full/export`
3. Make a simple change in the site's configuration, such as rearranging the fields on the Organism content type: `/admin/structure/bio_data/manage/organism/display`
5. Re-import the config archive: `/admin/config/development/configuration/full/import`
6. Do **not** get the following error message, (or any error message 🤞): `Configuration tripaldbx.settings depends on the tripaldbx extension that will not be installed after import.`, and apply the changes.
7. Navigate back to `/admin/structure/bio_data/manage/organism/display` to see that your changes made after the export (step 2) were reverted.
    

† @dsenalik had an issue where Docker kept files around that had been deleted during the build process. His workaround was to [use empty placeholder files](https://github.com/tripal/tripal/issues/2036#issuecomment-2494607062). If your Docker builds and the two deleted files remain, you'll also have to use placeholders.
